File: polyglot/app_yaml_overwrites/py/src/app_yaml_overwrites/overwrite_merger.py

from typing import Dict, Any

def apply_overwrites(original_config: Dict[str, Any], overwrite_section: Dict[str, Any]) -> Dict[str, Any]:
    """
    Deep merges overwrite_section into original_config.
    """
    if not overwrite_section:
        return original_config

    result = original_config.copy()
    
    for key, value in overwrite_section.items():
        if key in result and isinstance(result[key], dict) and isinstance(value, dict):
            result[key] = apply_overwrites(result[key], value)
        else:
            result[key] = value
            
    return result

File: polyglot/app_yaml_overwrites/mjs/src/overwrite-merger.ts
import _ from 'lodash';

export function applyOverwrites(originalConfig: any, overwriteSection: any): any {
    if (!overwriteSection) return originalConfig;

    // Deep merge using lodash, where overwriteSection takes precedence
    // Note: This matches the requirement to merge overwrites into parent
    // In a real implementation, we might need custom logic to handle 'null' specifically
    // if strict null handling is required vs undefined.
    // For this implementation, we assume standard merge.

    return _.merge({}, originalConfig, overwriteSection);
}
